src/searchModal.js

```diff
@@ -419,13 +433,50 @@ export default function searchModalFactory(config) {
      */
     function buildDataModel(data) {
         if (data.settings) {
+            //save availableColumns to memory
+            availableColumns = data.settings.availableColumns;
             // @todo: use the selected columns instead. It can use a promise as it takes place insise a promise chain
-            data.model = columnsToModel(data.settings.availableColumns);
+            data.model = columnsToModel(availableColumns);
+            dataCache = _.cloneDeep(data);
+            return data;
         } else {
             data.model = columnsToModel(_.values(data.model));
+            dataCache = _.cloneDeep(data);
+            return data;
         }
+    }
+
+    /**
+     * Filters datatble model based on stored selected columns
+     * @param {Object} data data containing available columns and model for datatable
+     * @returns {Promise} promise which resolves with filtered data.model
+     */
+    function filterSelectedColumns(data) {
+        return selectedColumnsStore
+            .getItem(getClassFilterUri())
+            .then(storedSelectedColumnIds => {
+                //let columnsToDisplay;
+                if (storedSelectedColumnIds && storedSelectedColumnIds.length) {
+                    selectedColumns = [...storedSelectedColumnIds];
+                    //columnsToDisplay = data.settings.availableColumns.filter(column =>
+                    // selectedColumns.includes(column.id)
+                    //);
+                } else {
+                    selectedColumns = data.settings.availableColumns.reduce((acc, column) => {
+                        if (column.default) {
+                            acc.push(column.id);
+                        }
+                    }, []);
```

Comment:
**issue:** The accumulator is never returned, and the browser will issue a runtime error: `TypeError: can't access property "push", acc is undefined`

**suggestion:**
```suggestion
                    selectedColumns = data.settings.availableColumns.reduce((acc, column) => {
                        if (column.default) {
                            acc.push(column.id);
                        }
                        return ACC;
                    }, []);
```

Reply:
Yes, right. I wonder how I've missed that..